### PR TITLE
Fixed Swift dynamic casting problem

### PR DESCRIPTION
**Pull request by jpsim**
**tl;dr; don't add your app's Swift files to your unit test targets.**

In Swift, classes are namespaced by the module (`ModuleName.ClassName`), which means that `RealmTest.TestUser` is technically a different class than `RealmTestTests.TestUser`. In turn, this causes Swift's dynamic casting (`as?`) to occasionally fail.

So instead of adding `TestUser.swift` to the unit test target (which causes it to be recompiled and two copies of the same class to be available), just make it `public` and import the module where it's being compiled into the test target.

To recap: in general, declare testable code written in Swift as `public` to expose it to your test targets, and don't recompile files with testable code in your test targets.

See a [note from the Swift testing framework Quick](https://github.com/Quick/Quick/blob/master/Documentation/SettingUpYourXcodeProject.md#testing-swift-code-using-swift) and [this StackOverflow comment](http://stackoverflow.com/questions/28733016/view-controller-tdd/28750266?noredirect=1#comment45781022_28736395) for more details.

### CODE REVIEW

**Comment by modocache** [comment]
:clap:

**Comment by bredfield** [comment]
Awesome. Thanks @jpsim.
